`apollo-server-types` dependency is missing in `apollo-datasource-rest`

When we try to generate our TS code without `skipLibCheck`, we have this error: `Cannot find module 'apollo-server-types' or its corresponding type declarations`
The .d.ts file generated by https://github.com/apollographql/apollo-server/blob/main/packages/apollo-datasource-rest/src/RESTDataSource.ts needs `apollo-server-types`.
We should declare `apollo-server-types` as dependency in the `apollo-datasource-rest` package.

```typescript
import { Request, RequestInit, Response, BodyInit, Headers, URL, URLSearchParams, URLSearchParamsInit, fetch } from 'apollo-server-env';
import type { ValueOrPromise } from 'apollo-server-types';
import { DataSource, DataSourceConfig } from 'apollo-datasource';
import { HTTPCache } from './HTTPCache';
import { ApolloError } from 'apollo-server-errors';
declare module 'apollo-server-env/dist/fetch' {
    interface RequestInit {
        cacheOptions?: CacheOptions | ((response: Response, request: Request) => CacheOptions | undefined);
    }
}
export declare type RequestOptions = RequestInit & {
    path: string;
    params: URLSearchParams;
    headers: Headers;
    body?: Body;
};
export interface CacheOptions {
    ttl?: number;
}
export declare type Body = BodyInit | object;
export { Request };
export declare abstract class RESTDataSource<TContext = any> extends DataSource {
    private httpFetch?;
    httpCache: HTTPCache;
    context: TContext;
    memoizedResults: Map<string, Promise<any>>;
    constructor(httpFetch?: typeof fetch | undefined);
    initialize(config: DataSourceConfig<TContext>): void;
    baseURL?: string;
    protected cacheKeyFor(request: Request): string;
    protected willSendRequest?(request: RequestOptions): ValueOrPromise<void>;
    protected resolveURL(request: RequestOptions): ValueOrPromise<URL>;
    protected cacheOptionsFor?(response: Response, request: Request): CacheOptions | undefined;
    protected didReceiveResponse<TResult = any>(response: Response, _request: Request): Promise<TResult>;
    protected didEncounterError(error: Error, _request: Request): void;
    protected parseBody(response: Response): Promise<object | string>;
    protected errorFromResponse(response: Response): Promise<ApolloError>;
    protected get<TResult = any>(path: string, params?: URLSearchParamsInit, init?: RequestInit): Promise<TResult>;
    protected post<TResult = any>(path: string, body?: Body, init?: RequestInit): Promise<TResult>;
    protected patch<TResult = any>(path: string, body?: Body, init?: RequestInit): Promise<TResult>;
    protected put<TResult = any>(path: string, body?: Body, init?: RequestInit): Promise<TResult>;
    protected delete<TResult = any>(path: string, params?: URLSearchParamsInit, init?: RequestInit): Promise<TResult>;
    private fetch;
    protected trace<TResult>(request: Request, fn: () => Promise<TResult>): Promise<TResult>;
}
//# sourceMappingURL=RESTDataSource.d.ts.map
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,7 @@ The version headers in this history reflect the versions of Apollo Server itself
 
 ## vNEXT
 
-- _Nothing yet! Stay tuned._
+- Add a few missing dependencies to packages. [PR #6393](https://github.com/apollographql/apollo-server/pull/6393)
 
 ## v3.7.0
 

--- a/packages/apollo-datasource-rest/package.json
+++ b/packages/apollo-datasource-rest/package.json
@@ -24,7 +24,7 @@
     "apollo-server-errors": "file:../apollo-server-errors",
     "http-cache-semantics": "^4.1.0"
   },
-  "devDependencies": {
-    "apollo-server-types": "file:../apollo-server-types"
+  "peerDependencies": {
+    "graphql": "^15.3.0 || ^16.0.0"
   }
 }

--- a/packages/apollo-datasource-rest/src/RESTDataSource.ts
+++ b/packages/apollo-datasource-rest/src/RESTDataSource.ts
@@ -10,8 +10,6 @@ import {
   fetch,
 } from 'apollo-server-env';
 
-import type { ValueOrPromise } from 'apollo-server-types';
-
 import { DataSource, DataSourceConfig } from 'apollo-datasource';
 
 import { HTTPCache } from './HTTPCache';
@@ -22,6 +20,7 @@ import {
   ForbiddenError,
 } from 'apollo-server-errors';
 
+type ValueOrPromise<T> = T | Promise<T>;
 declare module 'apollo-server-env/dist/fetch' {
   interface RequestInit {
     cacheOptions?:

--- a/packages/apollo-datasource-rest/tsconfig.json
+++ b/packages/apollo-datasource-rest/tsconfig.json
@@ -10,6 +10,5 @@
     { "path": "../apollo-datasource" },
     { "path": "../apollo-server-caching" },
     { "path": "../apollo-server-errors" },
-    { "path": "../apollo-server-types" },
   ]
 }

--- a/packages/apollo-server-types/package.json
+++ b/packages/apollo-server-types/package.json
@@ -11,6 +11,7 @@
     "node": ">=12.0"
   },
   "dependencies": {
+    "@apollo/utils.logger": "^1.0.0",
     "apollo-reporting-protobuf": "file:../apollo-reporting-protobuf",
     "apollo-server-caching": "file:../apollo-server-caching",
     "apollo-server-env": "file:../apollo-server-env"

--- a/packages/apollo-server/package.json
+++ b/packages/apollo-server/package.json
@@ -22,6 +22,7 @@
   },
   "homepage": "https://github.com/apollographql/apollo-server#readme",
   "dependencies": {
+    "@types/express": "4.17.13",
     "apollo-server-core": "file:../apollo-server-core",
     "apollo-server-express": "file:../apollo-server-express",
     "express": "^4.17.1"

